### PR TITLE
Add BlobSidecar by root RPC verification

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description
See https://github.com/ethereum/consensus-specs/pull/3535

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Review by StefanBratanov — APPROVED** [review]
LGTM

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/AbstractBlobSidecarsValidator.java [hunk 1]
```diff
@@ -20,19 +20,19 @@
...
   protected final KZG kzg;
 
-  public AbstractBlobSidecarsValidatingProxy(final Spec spec, final KZG kzg) {
+  public AbstractBlobSidecarsValidator(final Spec spec, final KZG kzg) {
     this.spec = spec;
     this.kzg = kzg;
   }
 
-  protected boolean verifyBlobSidecar(final BlobSidecar blobSidecar) {
+  boolean verifyBlobSidecarKzg(final BlobSidecar blobSidecar) {
```
Comment (StefanBratanov) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/AbstractBlobSidecarsValidator.java:35]: can rename to `validateBlobSidecarKzg` for consistency

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarByRootValidator.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import tech.pegasys.teku.kzg.KZG;
+import tech.pegasys.teku.networking.p2p.peer.Peer;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.datastructures.blobs.versions.deneb.BlobSidecar;
+import tech.pegasys.teku.spec.datastructures.networking.libp2p.rpc.BlobIdentifier;
+
+public class BlobSidecarByRootValidator extends AbstractBlobSidecarsValidator {
+
+  private final Peer peer;
+  private final Set<BlobIdentifier> blobIdentifiers;
```
Comment (StefanBratanov) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarByRootValidator.java]: can rename to `expectedBlobIdentifiers` to be more clear I think?

**Review by StefanBratanov — APPROVED** [review]
LGTM with couple of rename nits

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/AbstractBlobSidecarsValidator.java (hunk 1 above)
Comment (zilm13) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/AbstractBlobSidecarsValidator.java:35]: I was trying to make the most different name to validator to not mess up what to call from validator

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarByRootValidator.java (hunk 2 above)
Comment (zilm13) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarByRootValidator.java]: Was thinking on this, sure

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/AbstractBlobSidecarsValidator.java (hunk 1 above)
Comment (StefanBratanov) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/AbstractBlobSidecarsValidator.java:35]: sure, makes sense

**Pull request merged**